### PR TITLE
[ACCSEC-19361] - Added public method to clear local storage

### PR DESCRIPTION
**Pull request by sanarroya**
## Jira Ticket
- ACCSEC-19361

## Description
- Added public method to clear local storage

## Commit message
-  feat: Add public method to clear local storage

## Testing
- [x] Added unit tests
- [x] Ran unit tests successfully
- [x] Added documentation for public APIs and/or Wiki

### CODE REVIEW

**Comment by AuthyApps** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

## Code coverage
### Total coverage: **`95.56%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Data/Storage.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorFacade.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorRepository.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/PushFactory.swift | **`99.34%`**
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerifyManager.swift | **`100.00%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: TwilioVerifyTests/TwilioVerify/Sources/Integration/UpdateChallengeTests.swift [hunk 1]
```diff
@@ -1,157 +0,0 @@
...
-//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-//  See the License for the specific language governing permissions and
-//  limitations under the License.
-//
-
-import XCTest
-@testable import TwilioVerify
-
-// swiftlint:disable force_try
-class UpdateChallengeTests: BaseFactorTests {
```
Comment (yafuquen) [inline TwilioVerifyTests/TwilioVerify/Sources/Integration/UpdateChallengeTests.swift:24]: Why was this file deleted?

File: TwilioVerifyTests/TwilioVerify/Sources/Integration/ClearLocalStorageTests.swift [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+//  TwilioVerifyTests
+//
+//  Created by Santiago Avila on 11/3/20.
+//  Copyright © 2020 Twilio. All rights reserved.
+//
+
+import XCTest
+@testable import TwilioVerify
+
+class ClearLocalStorageTests: BaseFactorTests {
```
Comment (yafuquen) [inline TwilioVerifyTests/TwilioVerify/Sources/Integration/ClearLocalStorageTests.swift:12]: Do we need this test class?

File: README.md [hunk 3]
```diff
@@ -166,10 +167,19 @@ See [FactorListPresenter](https://github.com/twilio/twilio-verify-ios/blob/main/
...
   // Error
 }
 ```
+
+<a name='ClearLocalStorage'></a>
+
+## Clear local storage
+You can clear the local storage calling the `TwilioVerify.clearLocalStorage` method:
+```swift
+twilioVerify.clearLocalStorage()
```
Comment (yafuquen) [inline README.md:183]: Should we add `try`?

Comment (yafuquen) [reply README.md:183]: Can we have a method that doesn't throw an exception?

Comment (sanarroya) [reply README.md:183]: I thought about it and we can do it but it feel incomplete to me, I know that if it fails theres no much the consumer can do but I think we should at least give them the option.

PS I will add the `try` to the readme depending on what we decide about this

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.56%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Data/Storage.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorFacade.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorRepository.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/PushFactory.swift | **`99.34%`**
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerifyManager.swift | **`100.00%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: README.md (hunk 3 above)
Comment (yafuquen) [reply README.md:183]: Which errors could happen clearing the storage?

Comment (sanarroya) [reply README.md:183]: Keychain errors

Comment (yafuquen) [reply README.md:183]: Could you please investigate the possible errors?

Comment (yafuquen) [reply README.md:183]: I would like to know if an error could prevent the factors deletion. We are using the same method to delete one item, for one item an error could be item not found, but for this case we are deleting all items, so I am not sure the errors we can see

Comment (sanarroya) [reply README.md:183]: The errors are the same we can have when deleting a single factor

Comment (yafuquen) [reply README.md:183]: Can be an error thrown clearing local storage but adding a factor can work?

Comment (sanarroya) [reply README.md]: It can, I added a note on the README explaining why it could happen

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.56%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Data/Storage.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorFacade.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorRepository.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/PushFactory.swift | **`99.34%`**
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerifyManager.swift | **`100.00%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by yafuquen — APPROVED** [review]

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.55%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Data/Storage.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorFacade.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/FactorRepository.swift | **`100.00%`**
TwilioVerify/TwilioVerify/Sources/Domain/Factor/PushFactory.swift | **`99.33%`**
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerifyManager.swift | **`100.00%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by SergioFierro — APPROVED** [review]
